src/components/MultiHopTrade/components/TradeAmountInput.tsx

```diff
@@ -137,6 +165,20 @@ export const TradeAmountInput: React.FC<TradeAmountInputProps> = memo(
     const focusBg = useColorModeValue('gray.50', 'gray.900')
     const focusBorder = useColorModeValue('blue.500', 'blue.400')
 
+    const assetMarketDataUserCurrency = useAppSelector(state =>
+      selectMarketDataByAssetIdUserCurrency(state, assetId ?? ''),
+    )
+
+    // Local controller in case consumers don't have a form context, which is the case for all current consumers currentlyexcept RFOX
+    const _methods = useForm<TradeAmountInputFormValues>({
+      defaultValues: defaultFormValues,
+      mode: 'onChange',
+      shouldUnregister: true,
+    })
+    const methods = useFormContext<TradeAmountInputFormValues>()
+    const control = methods?.control ?? _methods.control
+    const setValue = methods?.setValue ?? _methods.setValue
```

Comment:
The magic bit that makes this a non-breaking change - failure to have a dummy local controller means this would throw when consumed outside of a form context